src/parse.rs

```diff
@@ -503,11 +503,11 @@ test!(strn =>
     pass "'str\\'3'";
     pass "\"str\\\"4\"";
 );
-build!(arrayaccess: Vec<WithOperators>, delimited(and!(openarr, optwhitespace), assignables, and!(optwhitespace, closearr)));
+build!(arrayaccess: Vec<Vec<WithOperators>>, delimited(and!(openarr, optwhitespace), assignablelist, and!(optwhitespace, closearr)));
 named_or!(varsegment: VarSegment =>
     Variable: String as variable,
     MethodSep: String as and!(optwhitespace, dot, optwhitespace),
-    ArrayAccess: Vec<WithOperators> as arrayaccess,
+    ArrayAccess: Vec<Vec<WithOperators>> as arrayaccess,
```

Comment:
Technically could have waited for another PR, but I included this to make sure the matrix-style access `matrix[1, 2]` works as desired.